sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_quickpulse/_exporter.py

```diff
@@ -244,31 +259,48 @@ def __init__(
         )
         self._worker.start()
 
+    # pylint: disable=protected-access
     def _ticker(self) -> None:
         if _is_ping_state():
             # Send a ping if elapsed number of request meets the threshold
             if self._elapsed_num_seconds % _get_global_quickpulse_state().value == 0:
-                ping_response = self._exporter._ping(  # pylint: disable=protected-access
+                ping_response = self._exporter._ping(
                     self._base_monitoring_data_point,
                 )
                 if ping_response:
-                    header = ping_response._response_headers.get(_QUICKPULSE_SUBSCRIBED_HEADER_NAME)  # pylint: disable=protected-access
-                    if header and header == "true":
-                        # Switch state to post if subscribed
-                        _set_global_quickpulse_state(_QuickpulseState.POST_SHORT)
-                        self._elapsed_num_seconds = 0
-                    else:
-                        # Backoff after _LONG_PING_INTERVAL_SECONDS (60s) of no successful requests
-                        if _get_global_quickpulse_state() is _QuickpulseState.PING_SHORT and \
-                            self._elapsed_num_seconds >= _LONG_PING_INTERVAL_SECONDS:
-                            _set_global_quickpulse_state(_QuickpulseState.PING_LONG)
+                    try:
+                        subscribed = ping_response._response_headers.get(_QUICKPULSE_SUBSCRIBED_HEADER_NAME)
+                        if subscribed and subscribed == "true":
+                            # Switch state to post if subscribed
+                            _set_global_quickpulse_state(_QuickpulseState.POST_SHORT)
+                            self._elapsed_num_seconds = 0
+                            # Update config etag
+                            etag = ping_response._response_headers.get(_QUICKPULSE_ETAG_HEADER_NAME)
+                            if etag is None:
+                                etag = ""
+                            if _get_quickpulse_etag() != etag:
+                                _set_quickpulse_etag(etag)
```

Comment:
I may be missing it - but let's say we go from post back to ping again. In that case, does the state for etag get set to empty string?

Reply:
actually I found it nvm

Reply:
In the current implementation, if the post request returns subscribed = False, the state goes back to pinging but the etag remains unchanged (not reset to an empty string). The exporter then pings with the old ("expired") etag. Is this not correct behavior?

Reply:
It isn't incorrect necessarily - from a functional perspective it doesn't matter what the etag is on a ping, as long as it exists (.net uses the last known etag on ping). In spec & in node.js code I do set it back to empty on ping, mainly because the etag doesn't mean much for pings. https://github.com/aep-health-and-standards/Telemetry-Collection-Spec/blob/main/ApplicationInsights/live-metrics-filtering.md#scenario-7-the-live-metrics-session-is-closed-after-metrics-have-been-emitting-for-a-while

Reply:
I have changed the behavior so etag is reset.